packages/zowe-explorer-api/src/tree/IZoweTree.ts

```diff
@@ -83,149 +85,176 @@ export interface IZoweTree<T> extends vscode.TreeDataProvider<T>, Partial<vscode
      * Add a new session to the container
      * @param zoweFileProvider The tree to which the profile should be added
      */
-    createZoweSession(zoweFileProvider: IZoweTree<IZoweTreeNode>): Promise<void>;
+    createZoweSession(zoweFileProvider: IZoweTree<IZoweTreeNode>): void | Promise<void>;
 
     /**
      * Create a brand new Schema
      * @param zoweFileProvider The tree from which the schema will be created
      */
-    createZoweSchema(zoweFileProvider: IZoweTree<IZoweTreeNode>): Promise<void>;
+    createZoweSchema(zoweFileProvider: IZoweTree<IZoweTreeNode>): void | Promise<void>;
 
     /**
      * Adds a favorite node
      * @param favorite Adds a favorite node
      */
-    checkCurrentProfile(node: IZoweTreeNode);
+    checkCurrentProfile(node: IZoweTreeNode): void | Promise<void>;
```

Comment:
Should this method return a type instead of void? `Promise<Validation.IValidationProfile>`

Also the doc string about adding a favorite doesn't seem right 😋